From d1212279fadfad50ffae3836cd5ae35888b7ca0b Mon Sep 17 00:00:00 2001
From: "Andre, Gerard" <gerard.andre@intel.com>
Date: Thu, 7 Apr 2016 10:11:45 +0100
Subject: [PATCH 2/5] Revert "Dollar Cove PMIC: add GPIO1 regulator."

This patch breaks the SD card detection.

This reverts commit bc806d8276afc32b22f8e3afe9cfd296356ef45f.
This reverts patch https://android.intel.com/#/c/483000

Change-Id: I5a5185762ac64945d3087c9f9368c2f6f3fc33a1
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26918
Reviewed-on: https://android.intel.com:443/491752
---
 drivers/mfd/intel_soc_pmic_dc.c | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index da77bba..25839e9 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -26,7 +26,6 @@
 #include <linux/power/dc_xpwr_battery.h>
 #include <linux/power/dc_xpwr_charger.h>
 #include <linux/regulator/intel_dollar_cove_pmic.h>
-#include <linux/regulator/machine.h>
 
 #include <asm/intel_em_config.h>
 #include <linux/extcon/extcon-dc-pwrsrc.h>
@@ -274,12 +273,6 @@ static struct mfd_cell dollar_cove_dev[] = {
 		.num_resources = 0,
 		.resources = NULL,
 	},
-	{
-		.name = "dcovex_regulator",
-		.id = DCOVEX_ID_GPIO1 + 1,
-		.num_resources = 0,
-		.resources = NULL,
-	},
 	{NULL, },
 };
 
@@ -667,28 +660,6 @@ static void dc_set_gpio_pdata(void)
 				sizeof(dollar_cove_gpio_data), 0);
 }
 
-static struct regulator_init_data dcovex_gpio1_data = {
-	.constraints = {
-		.name = "GPIO1",
-		.min_uV = 700000,
-		.max_uV = 3300000,
-		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE |
-					REGULATOR_CHANGE_STATUS,
-		.valid_modes_mask = REGULATOR_MODE_NORMAL,
-	},
-};
-
-static struct dcovex_regulator_info dollar_cove_reg_gpio1_data = {
-	.init_data = &dcovex_gpio1_data,
-};
-
-static void dc_set_regulator_pdata(void)
-{
-	intel_soc_pmic_set_pdata("dcovex_regulator",
-				(void *)&dollar_cove_reg_gpio1_data,
-				sizeof(dollar_cove_reg_gpio1_data),
-				DCOVEX_ID_GPIO1 + 1);
-}
 
 static int dollar_cove_init(void)
 {
@@ -697,7 +668,6 @@ static int dollar_cove_init(void)
 	dc_xpwr_pwrsrc_pdata();
 	dc_xpwr_fg_pdata();
 	dc_set_gpio_pdata();
-	dc_set_regulator_pdata();
 
 	return 0;
 }
-- 
1.9.1

